liquidctl/driver/nzxt_smart_device.py

```diff
@@ -163,29 +217,112 @@ def get_status(self, **kwargs):
         noise = []
-        for i, _ in enumerate(self._speed_channels):
-            msg = self.device.read(_READ_LENGTH)
-            LOGGER.debug('received %s', ' '.join(format(i, '02x') for i in msg))
-            num = (msg[15] >> 4) + 1
-            state = msg[15] & 0x3
-            status.append(('Fan {}'.format(num), ['—', 'DC', 'PWM'][state], ''))
-            noise.append(msg[1])
-            if state:
-                status.append(('Fan {} speed'.format(num), msg[3] << 8 | msg[4], 'rpm'))
-                status.append(('Fan {} voltage'.format(num), msg[7] + msg[8]/100, 'V'))
-                status.append(('Fan {} current'.format(num), msg[10]/100, 'A'))
-            if i != 0:
-                continue
-            fw = '{}.{}.{}'.format(msg[0xb], msg[0xc] << 8 | msg[0xd], msg[0xe])
-            status.append(('Firmware version', fw, ''))
-            if self._color_channels:
-                lcount = msg[0x11]
-                status.append(('LED accessories', lcount, ''))
-                if lcount > 0:
-                    ltype, lsize = [('Hue+ Strip', 10), ('Aer RGB', 8)][msg[0x10] >> 3]
-                    status.append(('LED accessory type', ltype, ''))
-                    status.append(('LED count (total)', lcount*lsize, ''))
-        status.append(('Noise level', round(sum(noise)/len(noise)), 'dB'))
+        
+        if self.device_type == self.DEVICE_SMARTDEV_V2:
+            numValidRepliesReceived = 0
+            msg1101Reply = False
+            msg2103Reply = False
+            msg6702Reply = False
+            msg6704Reply = False
+            # Get configuration information from device
+            wmsg = [0x10, 0x01, 0x00, 0x00, 0x00, 0x00]
+            LOGGER.info('Issuing command 0x10 0x01 to get firmware info')
+            self._write(wmsg)
+            wmsg = [0x20, 0x03, 0x00, 0x00, 0x00, 0x00]
+            LOGGER.info('Issuing command 0x20 0x03 to get lighting info')
+            self._write(wmsg)
+            #wmsg = [0x60, 0x03, 0x00, 0x00, 0x00, 0x00]
+            #LOGGER.info('Issuing command 0x20 0x03 to get lighting info')
+            #self._write(wmsg)
+            # After issuing the above 2 commands, we will get a series of replies that
+            # will include everything we want to extract and display to the user.
+            # It may take 10 or 12 reply messages before we get all of the expected replies.
+            for x in range(12):     # we may not get reply right away, so try up to 12 times
+                if numValidRepliesReceived == 4:
+                    break   
+                msg = self.device.read(_READ_LENGTH_V2)
+                LOGGER.debug('received %s', ' '.join(format(i, '02x') for i in msg))
+                if msg1101Reply == False and msg[0] == 0x11 and msg[1] == 0x01:  # the correct reply must start with 0x11, 0x01
+                    fw = '{}.{}.{}'.format(msg[0x11], msg[0x12], msg[0x13])
+                    status.append(('Firmware:', fw, ''))
```

Comment:
Sorry, only noticed this now: please use `'Firmware version'` (no colon) as the key, for uniformity with the other drivers.

Reply:
Done!